src/hopen/syntax/silly_j.cljc

```diff
@@ -0,0 +1,51 @@
+(ns hopen.syntax.silly-j
+  (:require [hopen.renderer.xf :as rxf]
+            [instaparse.core :as insta]))
+
+(declare parse)
+
+;; TODO test case
+(defn render
+  ([template data]
+   (render template data '()))
+  ([template data other-env]
+   (let [new-env (update rxf/default-env :bindings assoc other-env)
+         renderer (rxf/renderer (parse template) new-env)]
+     (transduce renderer str data))))
+
+(def ebnf
+  (insta/parser "
+                S       = [STR | SILLYSTR]+
+                STR      = #'[^{}]+'
+                SILLYSTR =  OBK SILLY CBK
+                OBK      = '{'
+                CBK      = '}'
```

Comment:
I would suggest using non-abbreviated names or less abbreviated names, to ensure that anybody can easily read the source code.

Reply:
As a side note: the casing (upper-case / lower-case) does not matter for instaparse.